Atomic/containers.py

```diff
@@ -289,7 +290,70 @@ def _mark_vulnerable(self, containers):
             if con.id in vulnerable_uuids:
                 con.vulnerable = True
 
+    def update_all_containers(self):
+        could_not_pull = {}
+        preupdate_containers = self.syscontainers.get_containers()
+        for i in preupdate_containers:
+            img = i['Image']
+            pull_location = i['PullLocation']
+            name = i['Names'][0]
+            d = util.Decompose(img)
+            if d.registry == "":
+                util.write_out("Image {} not fully qualified: skipping {}".format(img, name))
+                continue
+            try:
+                self.syscontainers.pull_image(pull_location)
+            except:  # pylint: disable=bare-except
+                could_not_pull[img] = True
+                util.write_out("Image {} could not be pulled.  You may need to manually update {}".format(img, name))
+                continue
+
+            self.syscontainers.update_container(name, controlled=True)
+        postupdate_containers = self.syscontainers.get_containers()
+
+        images_by_name = {i['RepoTags'][0] : i for i in self.syscontainers.get_system_images()}
+        preupdate_containers_by_name = {i['Names'][0] : i for i in preupdate_containers}
+        postupdate_containers_by_name = {i['Names'][0] : i for i in postupdate_containers}
+
+        def get_image_id(c): return c['ImageId'] if 'ImageId' in c else c['ImageID']
+
+        def get_status(img_name, pre_id, post_id, image_id):
+            if img_name in could_not_pull.keys():
+                return "Could not pull"
+            if image_id == None:
+                return "Unknown"
+            if post_id == image_id and image_id != pre_id:
+                return "Updated now"
+            if post_id == image_id and image_id == pre_id:
+                return "Updated"
+            return "Not updated"
+
+        cols = "{0:15} {1:32} {2:32} {3:32} {4:15}"
+
+        util.write_out("\nSUMMARY\n")
+        util.write_out(cols.format("Container", "Image ID before update", "Image ID after update", "Latest image ID", "Status"))
+        for cnt in preupdate_containers_by_name.keys():
+            pre_version = get_image_id(preupdate_containers_by_name[cnt])
+            post_version = get_image_id(postupdate_containers_by_name[cnt])
+            img_name = img_id = None
+            try:
+                img_name = preupdate_containers_by_name[cnt]['Image']
+                img_id = get_image_id(images_by_name[img_name])
+            except KeyError:
+                pass
+            status = get_status(img_name, pre_version, post_version, img_id)
+            util.write_out(cols.format(cnt[:15], pre_version[:32], post_version[:32], (img_id or "<NOT FOUND>")[:32], status[:15]))
+
     def update(self):
+        if self.args.all:
```

Comment:
Should you throw an error if self.args.all is set and self.args.container is not None?